middleware/session/session.go

```diff
@@ -171,6 +171,11 @@ func (s *Session) setCookie() {
 
 	s.ctx.Response().Header.SetCookie(fcookie)
 	fasthttp.ReleaseCookie(fcookie)
+
+	if s.config.Source == SourceHeader {
+		s.ctx.Request().Header.SetBytesV(s.config.CookieName, []byte(s.id))
+		s.ctx.Response().Header.SetBytesV(s.config.CookieName, []byte(s.id))
+	}
 }
 
 func (s *Session) delCookie() {
```

Comment:
I think we should have separate functions for extracting a session ID from a cookie and from a header. You can find an example of this in our [csrf middleware](https://github.com/gofiber/fiber/blob/master/middleware/csrf/csrf.go#L30-L39).